src/qutip_qip/circuit/mat_renderer.py

```diff
@@ -0,0 +1,936 @@
+"""
+Module for rendering a quantum circuit using matplotlib library.
+"""
+
+from typing import List
+
+import numpy as np
+import matplotlib.pyplot as plt
+from matplotlib.patches import (
+    FancyBboxPatch,
+    Circle,
+    Arc,
+    FancyArrow,
+)
+
+from ..operations import Gate, Measurement
+from ..circuit import QubitCircuit
+from .color_theme import default_theme
+
+__all__ = [
+    "MatRenderer",
+]
+
+
+class MatRenderer:
+    """
+    Class to render a quantum circuit using matplotlib.
+
+    Parameters
+    ----------
+    qc : QuantumCircuit Object
+        The quantum circuit to be rendered.
+
+    padding : float, optional
+        The padding around the circuit. The default is 0.3.
+
+    dpi : int, optional
+        The resolution of the image. The default is 150.
+
+    bgcolor : str, optional
+        The background color of the plotted circuit. The default is "#FFFFFF" (white).
+
+    wire_labels : list, optional
+        The labels for the wires.
+    """
+
+    def __init__(
+        self,
+        qc: QubitCircuit,
+        padding=0.3,
+        dpi: int = 150,
+        bgcolor: str = "#FFFFFF",
+        wire_label: List[str] = None,
+        condense=0.15,
+        bulge=True,
+    ) -> None:
+
+        self.wire_sep = 0.5
+        self.layer_sep = 0.5
+        self.cwire_sep = 0.02
+        self.gate_height = 0.2
+        self.gate_width = 0.2
+        self.gate_pad = 0.05
+        self.label_pad = 0.2
+        self.font_size = 10
+        self.default_layers = 2
+        self.arrow_lenght = 0.06
+        self.connector_r = 0.01
+        self.target_node_r = 0.12
+        self.display_layer_len = 0
+        self.start_pad = 0.1
```

Comment:
Would it be cleaner to separate the properties about plotting configuration into a separate dict or data class?
Those are the properties that we allow the use to play with.
This will distinguish them from the other properties like `self.qc`, `self.qwires`, which I believe is should not be changed by the user.